app/src/main/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivity.java

```diff
@@ -52,6 +54,8 @@
 public class ActivityDescriptionActivity extends AppCompatActivity {
 
     private static final String TAG = "ActDescActivity";
+    public static final String DESCRIPTION_ACTIVITY_KEY = "activitykey";
```

Comment:
Oh, this is a good practice to define the constant strings for retrieving values from intents in as static fields at the beginning of the class, I will apply this in ChatActivity and EditProfileActivity!

But why did you set it as public? Do you plan to use this String in another class?

And did you also modify this hard-coded String in MainMapFragment (i.e. the activity from which the intent containing the "activity" object comes) in the same way?

Reply:
Yes, it is also used in the GPSRecording class, when returning from a recording. This way, when recreating the ActivityDescriptionActivity, it can use the newly recorded data passed in the updated Activity.

Yes, all interactions which required passing an Activity to the ActivityDescriptionActivity should have been updated with this constant.